newIDE/app/src/EventsSheet/EventsTree/Renderers/CommentEvent.js

```diff
@@ -94,48 +89,80 @@ export default class CommentEvent extends React.Component<
-    );
+    var commentEvent = gd.asCommentEvent(this.props.event);
+
+    const r = commentEvent.getBackgroundColorRed(),
+      g = commentEvent.getBackgroundColorGreen(),
+      b = commentEvent.getBackgroundColorBlue();
+
+    const textColors = {
+        red: commentEvent.getTextColorRed(),
+        green: commentEvent.getTextColorGreen(),
+        blue: commentEvent.getTextColorBlue(),
+    };
+
+    const textColor = `rgb(${textColors.red}, ${textColors.green}, ${textColors.blue})`;
 
     return (
-      <div
-        style={{ ...styles.container, backgroundColor: `#${color}` }}
-        className={classNames({
-          [largeSelectableArea]: true,
-          [largeSelectedArea]: this.props.selected,
-        })}
-        ref={container => (this._container = container)}
-      >
-        {!this.state.editing ? (
-          <p
-            className={classNames({
-              [selectableArea]: true,
-            })}
-            onClick={this.edit}
-            key="p"
-            style={{ ...styles.text, color: `#${textColor}` }}
-            dangerouslySetInnerHTML={{
-              __html: this._getCommentHTML(),
-            }}
-          />
-        ) : (
-          <textarea
-            key="textarea"
-            type="text"
-            style={{ ...styles.textArea, height: this.state.height }}
-            onBlur={this.endEditing}
-            ref={input => (this._input = input)}
-          />
-        )}
+      <div>
+        <div
+          className={classNames({
+            [largeSelectableArea]: true,
+            [largeSelectedArea]: this.props.selected,
+          })}
+          style={{
+            ...styles.container,
+            backgroundColor: `rgb(${r}, ${g}, ${b})`,
+            height: this.state.height,
+            minHeight: 35,
+          }}
+          onClick={this.edit}
+        >
+          {this.state.editing ? (
+            <TextField
+              multiLine={true}
+              margin="none"
+              ref={textField => (this._textField = textField)}
+              value={commentEvent.getComment()}
+              hintText={t`<Enter comment>`}
+              onBlur={this.endEditing}
+              onChange={this.onEvent}
+              style={{ ...styles.textField, padding: 5 }}
+              inputStyle={{
+                color: textColor,
+                WebkitTextFillColor: textColor,
+                padding: 0,
+                lineHeight: '1.5em',
+              }}
+              underlineFocusStyle={{
+                borderColor: textColor,
+              }}
+              fullWidth
+              id="comment-title"
+            />
+          ) : (
+            <span
+              ref={selectable => (this._selectable = selectable)}
+              className={classNames({
+                [selectableArea]: true,
+                [disabledText]: this.props.disabled,
+              })}
+              style={{
+                ...styles.textField,
```

Comment:
Not a big fan of reusing styles called `textField` in something that is not a textField. This makes the reader and the future maintainer of this code less confident about what this style are used for and if they will break something when changed. Instead, just put the whole style in a single object in `styles`: `style={styles.commentSpan}`